plugins/infinispan4/src/main/java/org/radargun/service/InfinispanEmbeddedService.java

```diff
@@ -33,7 +33,7 @@
 import static java.util.concurrent.TimeUnit.MINUTES;
 
 @Service(doc = InfinispanEmbeddedService.SERVICE_DESCRIPTION)
-public class InfinispanEmbeddedService {
+public class InfinispanEmbeddedService implements InfinispanTransactionalService {
    protected static final String SERVICE_DESCRIPTION = "Service hosting Infinispan in embedded (library) mode.";
```

Comment:
@diegolovison I think this approach is wrong. You should add the `Transactional` trait to the implementation of the HotRodService for 9.3 or 9.4. (I can't remember which version this was added in.) @rvansa isn't that correct?

Reply:
@diegolovison did expose the `Transactional` trait but rather than creating his own implementation he reused the embedded one, and added some helpers through the InfinispanTransactionalService. I haven't undersood the changes completely but I must say that I dislike the `if (... instanceof HotRodService)` in Infinispan94Transactional. Hot Rod and Embedded implementations should not share the implemenation in RG unless the implementation in Infinispan is really common.

Reply:
Hot Rod Transaction is supported since 9.3. We should create a new plugin `infinispan93` and move the Hot Rod transaction support into `infinispan93`

Hot Rod and Embedded Transaction is the same from the client perspective.
https://github.com/radargun/radargun/blob/387c90a53adf83b99993b3ba7b3fbfec535b836a/plugins/infinispan4/src/main/java/org/radargun/service/InfinispanTransactional.java#L46